src/makeCalculator.js

```diff
@@ -4,7 +4,39 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {};
+
+  calculator.result = 0;
+
+  calculator.add = (num) => {
+    calculator.result += num;
+  };
+
+  calculator.subtract = (num) => {
+    calculator.result -= num;
+  };
+
+  calculator.multiply = (num) => {
+    calculator.result *= num;
+  };
+
+  calculator.divide = (num) => {
+    calculator.result /= num;
+  };
+
+  calculator.reset = () => {
+    calculator.result = 0;
+
+    return calculator;
+  };
+
+  calculator.operate = (callback, num) => {
+    callback(num);
+
+    return calculator;
+  };
```

Comment:
In the `operate` method, it's a good practice to check if `callback` is a function before invoking it. This can prevent runtime errors if an invalid argument is passed.